src/inference/dev_api/openvino/runtime/icore.hpp

```diff
@@ -0,0 +1,251 @@
+     *
+     * Users can create as many networks as they need and use
+     *        them simultaneously (up to the limitation of the hardware resources)
+     *
+     * @param model_str String data of model
+     * @param weights Model's weights
+     * @param device_name Name of device to load network to
+     * @param config Optional map of pairs: (config parameter name, config parameter value) relevant only for this load
+     * operation
+     * @return An executable network reference
+     */
+    virtual ov::SoPtr<InferenceEngine::IExecutableNetworkInternal> compile_model(const std::string& model_str,
+                                                                                 const ov::Tensor& weights,
+                                                                                 const std::string& device_name,
+                                                                                 const ov::AnyMap& config) const = 0;
+
+    /**
+     * @brief Creates an executable network from a previously exported network
+     * @param model model stream
+     * @param device_name Name of device load executable network on
+     * @param config Optional map of pairs: (config parameter name, config parameter value) relevant only for this load
+     * operation*
+     * @return An executable network reference
+     */
+    virtual ov::SoPtr<InferenceEngine::IExecutableNetworkInternal>
+    import_model(std::istream& model, const std::string& device_name = {}, const ov::AnyMap& config = {}) const = 0;
+
+    /**
+     * @brief Query device if it supports specified network with specified configuration
+     *
+     * @param model OpenVINO Model
+     * @param device_name A name of a device to query
+     * @param config Optional map of pairs: (config parameter name, config parameter value)
+     * @return An object containing a map of pairs a layer name -> a device name supporting this layer.
+     */
+    virtual ov::SupportedOpsMap query_model(const std::shared_ptr<const ov::Model>& model,
+                                            const std::string& device_name,
+                                            const ov::AnyMap& config) const = 0;
+
+    /**
+     * @brief Returns devices available for neural networks inference
+     *
+     * @return A vector of devices. The devices are returned as { CPU, GPU.0, GPU.1, MYRIAD }
+     * If there more than one device of specific type, they are enumerated with .# suffix.
+     */
+    virtual std::vector<std::string> get_available_devices() const = 0;
+
+    /**
+     * @brief Checks whether device supports Export & Import functionality of network
+     *
+     * @param device_name - A name of a device to get a metric value.
+     * @return True if device has IMPORT_EXPORT_SUPPORT metric in SUPPORTED_METRICS and
+     * this metric returns 'true', False otherwise.
+     */
+    virtual bool device_supports_import_export(const std::string& device_name) const = 0;
```

Comment:
why this method is pure virtual? I suppose we can easily provide a default implementation. 
Secondly, I see it's used only inside CoreImpl as a helper method, so likely it should move out of ICore, which represents API between plugin and Core

the same for get_supported_property.

About set_property - I suppose we don't need to provide a mechanism for plugin to change properties for other plugins.

Reply:
Removed `device_supports_import_export`, `get_supported_property` and `set_property` from the interface.